app/bundles/EmailBundle/Swiftmailer/Transport/AmazonApiTransport.php

```diff
@@ -0,0 +1,842 @@
+            $tokens       = (!empty($metadataSet['tokens'])) ? $metadataSet['tokens'] : [];
+            $mauticTokens = array_keys($tokens);
+        }
+
+        foreach ($metadata as $recipient => $mailData) {
+            $this->message->setBody($emailBody);
+            $msg        = $this->messageToArray($mauticTokens, $mailData['tokens'], true);
+            $rawMessage = $this->buildRawMessage($msg, $recipient);
+            $payload    = [
+                'Source'       => $msg['from']['email'],
+                'Destinations' => [$recipient],
+                'RawMessage'   => [
+                    'Data' => $rawMessage,
+                    ],
+            ];
+
+            yield $payload;
+        }
+    }
+
+    /**
+     * @param type $msg
+     * @param type $recipient
+     *
+     * @return string
+     */
+    public function buildRawMessage($msg, $recipient)
+    {
+        $separator           = md5(time());
+        $separator_multipart = md5($msg['subject'].time());
+        $message             = "MIME-Version: 1.0\n";
+        $message .= 'Subject: '.$msg['subject']."\n";
+        $message .= 'From: '.$msg['from']['name'].' <'.$msg['from']['email'].">\n";
+        $message .= "To: $recipient\n";
+        if (count($msg['recipients']['cc']) > 0) {
+            $message .= 'Cc: '.implode(',', array_keys($msg['recipients']['cc']))."\n";
+        }
+        if (count($msg['recipients']['bcc']) > 0) {
+            $message .= 'Bcc: '.implode(',', array_keys($msg['recipients']['bcc']))."\n";
+        }
+        $message .= "Content-Type: multipart/mixed; boundary=\"$separator_multipart\"\n";
+        $message .= "\n--$separator_multipart\n";
+
+        $message .= "Content-Type: multipart/alternative; boundary=\"$separator\"\n";
+        if (isset($msg['text']) && strlen($msg['text']) > 0) {
+            $message .= "\n--$separator\n";
+            $message .= "Content-Type: text/plain; charset=\"UTF-8\"\n";
+            $message .= "Content-Transfer-Encoding: base64\n";
+            $message .= "\n".wordwrap(base64_encode($msg['text']), 78, "\n", true)."\n";
+        }
+        $message .= "\n--$separator\n";
+        $message .= "Content-Type: text/html; charset=\"UTF-8\"\n";
+        $message .= "\n".$msg['html']."\n";
+        $message .= "\n--$separator--\n";
+
+        foreach ($msg['attachments'] as $attachment) {
+            $message .= "--$separator_multipart\n";
+            $message .= 'Content-Type: '.$attachment['type'].'; name="'.$attachment['name']."\"\n";
+            $message .= 'Content-Disposition: attachment; filename="'.$attachment['name']."\"\n";
+            $message .= "Content-Transfer-Encoding: base64\n";
+            $message .= "\n".$attachment['content']."\n";
+        }
+
+        $message .= "--$separator_multipart--";
+
+        return $message;
+    }
+
+    /**
+     * @return int
+     */
+    public function getMaxBatchLimit()
+    {
+        return 10;
```

Comment:
I'm not sure if this 10 is a good choice, if it was based on AWS limits or just a hard number. Templated messages can be sent to 50 destinations. I increased it in my production deployment to increase my send rate. Maybe 10 is the limit only for raw messages?

https://docs.aws.amazon.com/ses/latest/DeveloperGuide/limits.html